### ISSUE TITLE
future of this package

### ISSUE THREAD

**Issue by tpapp** [JuliaArrays/EndpointRanges.jl#21]
After we got `begin` in JuliaLang#33946, I am wondering if this package is necessary. Maybe include a notice in the readme.

**Comment by roflmaostc** [comment]
There was already a discussion in #15.
But as @timholy mentioned, the following doesn't really work with Base:
```julia
julia> using EndpointRanges

julia> a = [1,2,3,4,5]
5-element Vector{Int64}:
 1
 2
 3
 4
 5

julia> view(a, begin:2)
ERROR: syntax: "begin" at REPL[29]:1 expected "end", got ")"
Stacktrace:
 [1] top-level scope
   @ none:1

julia> view(a, ibegin:2)
2-element view(::Vector{Int64}, 1:2) with eltype Int64:
 1
 2
 ```
 
 So yeah, the use-case got narrower, but the gap isn't closed by Base (yet?).

**Comment by tpapp** [comment]
But isn't that what
```julia
@view a[begin+2:end-3]
```
is for?

Please don't get me wrong, I found this package very useful in the past, but I am genuinely curious what @timholy thinks about its future.

I think that the biggest weakness of this approach is having to define functions for operations on `IndexFunction`. Eg suppose a user wants `isqrt(ibegin + 2)`, is that a reasonable request?

**Comment by roflmaostc** [comment]
Yeah I agree on the issue that you basically need to redefine all numerical functions like `sqrt` on the type.

But on the other hand, it allows to define quickly something like `imid` which Base doesn't offer currently.
Users even could define different indices like `firstprimeindex`, couldn't they?
